src/commonMain/kotlin/baaahs/shows/ThumpShow.kt

```diff
@@ -0,0 +1,68 @@
+package baaahs.shows
+
+import baaahs.*
+import baaahs.shaders.CompositingMode
+import baaahs.shaders.CompositorShaderBuffer
+import baaahs.shaders.SineWaveShaderBuffer
+import baaahs.shaders.SolidShaderBuffer
+import kotlin.math.PI
+import kotlin.random.Random
+
+class ThumpShow(sheepModel: SheepModel, showRunner: ShowRunner) : Show {
+    private val beatProvider = showRunner.getBeatProvider()
+    private val colorPicker = showRunner.getColorPicker()
+
+    private val shaderBufs = sheepModel.allPanels.map { panel ->
+        val solidShader = showRunner.getSolidShader(panel)
+
+        val sineWaveShader = showRunner.getSineWaveShader(panel).apply {
+            buffer.density = Random.nextFloat() * 20
+        }
+
+        val compositorShader = showRunner.getCompositorShader(panel, solidShader, sineWaveShader)
+
+        compositorShader.buffer.apply {
+            mode = CompositingMode.ADD
+            fade = 1f
+        }
+
+        ShaderBufs(solidShader.buffer, sineWaveShader.buffer, compositorShader.buffer)
+    }
+
+    private val movingHeadBuffers = sheepModel.eyes.map { showRunner.getMovingHead(it) }
+
+    init {
+//        println("Created new CompositeShow, we have ${shaderBufs.size} buffers")
+    }
+
+    override fun nextFrame() {
+        val theta = ((getTimeMillis() / 1000f) % (2 * PI)).toFloat()
+        val beat = beatProvider.beat
+
+        var i = 0
+        shaderBufs.forEach { shaderBuffer ->
+            shaderBuffer.solidShaderBuffer.color = Color.BLACK.fade(colorPicker.color, beat % 1f)
+            shaderBuffer.sineWaveShaderBuffer.color = if (beat < .2) Color.WHITE else Color.ORANGE
```

Comment:
Guessing beat would be .1, .2, .3, .4 and then wraps back to .1 every bar?

Reply:
I was thinking 0.0, 1.0, 2.0, 3.0 (for 4/4 time), but it's a float so you get intermediate values too.